Go: TestAddChainRetries is flaky

```
=== RUN   TestAddChainRetries

2017/01/30 19:19:49 Request failed, backing-off for 1s: got HTTP status 503 Service Unavailable

2017/01/30 19:19:50 Request failed, backing-off for 2s: got HTTP status 503 Service Unavailable

2017/01/30 19:19:52 Request failed, backing-off for 4s: got HTTP status 503 Service Unavailable

2017/01/30 19:19:56 Request failed, backing-off for 5s: got HTTP status 503 Service Unavailable

2017/01/30 19:20:01 Request failed, backing-off for 10s: got HTTP status 503 Service Unavailable

2017/01/30 19:20:06 Request failed, backing-off for 1s: got HTTP status 503 Service Unavailable

2017/01/30 19:20:07 Request failed, backing-off for 1s: got HTTP status 503 Service Unavailable

2017/01/30 19:20:08 Request failed, backing-off for 1s: got HTTP status 503 Service Unavailable

2017/01/30 19:20:09 Request failed, backing-off for 1s: got HTTP status 503 Service Unavailable

2017/01/30 19:20:10 Request failed, backing-off for 1s: got HTTP status 503 Service Unavailable

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

2017/01/30 19:20:11 Request timed out, retrying immediately

--- FAIL: TestAddChainRetries (22.14s)

	logclient_test.go:262: #4 Submission took an unexpected length of time: 5.111453326s, expected ~5s

```

--- a/go/client/logclient_test.go
+++ b/go/client/logclient_test.go
@@ -227,6 +227,7 @@ func TestAddChainRetries(t *testing.T) {
 		t.Fatalf("Failed to create client: %v", err)
 	}
 	leeway := time.Millisecond * 100
+	leewayRatio := 0.1 // 10%
 	instant := time.Millisecond
 	fiveSeconds := time.Second * 5
 	sevenSeconds := time.Second * 7 // = 1 + 2 + 4
@@ -258,7 +259,9 @@ func TestAddChainRetries(t *testing.T) {
 		started := time.Now()
 		sct, err := client.AddChain(deadline, chain)
 		took := time.Since(started)
-		if math.Abs(float64(took-test.expected)) > float64(leeway) {
+		delta := math.Abs(float64(took - test.expected))
+		ratio := delta / float64(test.expected.Nanoseconds())
+		if delta > float64(leeway) && ratio > leewayRatio {
 			t.Errorf("#%d Submission took an unexpected length of time: %s, expected ~%s", i, took, test.expected)
 		}
 		if test.success && err != nil {

--- a/go/client/logclient_test.go
+++ b/go/client/logclient_test.go
@@ -226,11 +226,11 @@ func TestAddChainRetries(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to create client: %v", err)
 	}
-	leeway := time.Millisecond * 100
-	leewayRatio := 0.1 // 10%
-	instant := time.Millisecond
-	fiveSeconds := time.Second * 5
-	sevenSeconds := time.Second * 7 // = 1 + 2 + 4
+	const leeway = time.Millisecond * 100
+	const leewayRatio = 0.1 // 10%
+	const instant = time.Millisecond
+	const fiveSeconds = time.Second * 5
+	const sevenSeconds = time.Second * 7 // = 1 + 2 + 4
 
 	tests := []struct {
 		deadlineLength        int
@@ -260,7 +260,7 @@ func TestAddChainRetries(t *testing.T) {
 		sct, err := client.AddChain(deadline, chain)
 		took := time.Since(started)
 		delta := math.Abs(float64(took - test.expected))
-		ratio := delta / float64(test.expected.Nanoseconds())
+		ratio := delta / float64(test.expected)
 		if delta > float64(leeway) && ratio > leewayRatio {
 			t.Errorf("#%d Submission took an unexpected length of time: %s, expected ~%s", i, took, test.expected)
 		}

